### DISCUSSION TITLE
Parallel rendering support in MJX?

### DISCUSSION THREAD

**Discussion by StoneT2000** [google-deepmind/mujoco#1343] [General]
I saw there were some other posts (https://github.com/google-deepmind/mujoco/discussions/1101#discussioncomment-7315694, https://github.com/google-deepmind/mujoco/issues/1264) but couldn't seem to see if there was a clear solution. I would like to train on some mjx powered tasks with GPU simulation in addition to rendering every parallel environment for visual-based RL. Is this possible and if not, are there any plans to support this?

Thanks!

**Comment by erikfrey** [comment]
Hi there - on HEAD you can now find `mjx.ray` which mirrors [mj_ray](https://mujoco.readthedocs.io/en/latest/APIreference/APIfunctions.html#mj-ray) for planes, spheres, capsules, boxes, and meshes, so you can simulate LiDAR.  For simple scenes it should be quite performant.

For RGB, there's nothing currently on the MJX roadmap, but we would love to support such an effort.  MJX can interoperate with other GPU programs via [dlpack](https://github.com/dmlc/dlpack) and there seem to be a number of python 3d rendering libraries that this might work with, such as [Open3D](https://www.open3d.org/docs/latest/tutorial/core/tensor.html#PyTorch-I/O-with-DLPack-memory-map).

If anyone would like to try this, please let us know!

**Reply by Andrew-Luo1** [reply]
mjx.ray indeed appears to be quite performant. I discuss a simple depth camera implementation [here](https://github.com/google-deepmind/mujoco/issues/1485), which renders up to several million FPS.

**Comment by dantes-ai** [comment]
Interested in this as well. This currently means that MJX does not support scaling up visual-based RL...
